### PR TITLE
Azure Sphere .NET SDK 

### PR DESCRIPTION
**Pull request by allegradomel**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/sphere/Azure.ResourceManager.Sphere/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Microsoft Azure Sphere management client library for .NET
+
+**[Describe the service briefly first.]**
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/README.md:3]: Please put some description for your service here

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs [hunk 2]
```diff
@@ -0,0 +1,611 @@
...
+        public virtual Azure.AsyncPageable<Azure.ResourceManager.Sphere.CatalogResource> GetAllAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Sphere.CatalogResource>> GetAsync(string catalogName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        System.Collections.Generic.IAsyncEnumerator<Azure.ResourceManager.Sphere.CatalogResource> System.Collections.Generic.IAsyncEnumerable<Azure.ResourceManager.Sphere.CatalogResource>.GetAsyncEnumerator(System.Threading.CancellationToken cancellationToken) { throw null; }
+        System.Collections.Generic.IEnumerator<Azure.ResourceManager.Sphere.CatalogResource> System.Collections.Generic.IEnumerable<Azure.ResourceManager.Sphere.CatalogResource>.GetEnumerator() { throw null; }
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
+    }
+    public partial class CatalogData : Azure.ResourceManager.Models.TrackedResourceData
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs:18]: Could we rename this class to `ShpereCatalogData`?

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs [hunk 3]
```diff
@@ -0,0 +1,611 @@
...
+    {
+        public ProofOfPossessionNonceContent(string proofOfPossessionNonce) { }
+        public string ProofOfPossessionNonce { get { throw null; } }
+    }
+    public partial class ProofOfPossessionNonceResponse : Azure.ResourceManager.Sphere.Models.CertificateProperties
+    {
+        internal ProofOfPossessionNonceResponse() { }
+    }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct ProvisioningState : System.IEquatable<Azure.ResourceManager.Sphere.Models.ProvisioningState>
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs:548]: Could we rename this type to `SphereProvisioningState`?

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs [hunk 4]
```diff
@@ -0,0 +1,611 @@
...
+        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Sphere.CertificateResource>> GetAsync(string serialNumber, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        System.Collections.Generic.IAsyncEnumerator<Azure.ResourceManager.Sphere.CertificateResource> System.Collections.Generic.IAsyncEnumerable<Azure.ResourceManager.Sphere.CertificateResource>.GetAsyncEnumerator(System.Threading.CancellationToken cancellationToken) { throw null; }
+        System.Collections.Generic.IEnumerator<Azure.ResourceManager.Sphere.CertificateResource> System.Collections.Generic.IEnumerable<Azure.ResourceManager.Sphere.CertificateResource>.GetEnumerator() { throw null; }
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
+    }
+    public partial class CertificateData : Azure.ResourceManager.Models.ResourceData
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs:75]: Could we rename this type to `SphereCertificateData`?

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs [hunk 5]
```diff
@@ -0,0 +1,611 @@
...
+        public string Certificate { get { throw null; } }
+        public System.DateTimeOffset? ExpiryUtc { get { throw null; } }
+        public System.DateTimeOffset? NotBeforeUtc { get { throw null; } }
+        public Azure.ResourceManager.Sphere.Models.ProvisioningState? ProvisioningState { get { throw null; } }
+        public Azure.ResourceManager.Sphere.Models.CertificateStatus? Status { get { throw null; } }
+        public string Subject { get { throw null; } }
+        public string Thumbprint { get { throw null; } }
+    }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct CertificateStatus : System.IEquatable<Azure.ResourceManager.Sphere.Models.CertificateStatus>
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs:406]: Could we rename this type to `SphereCertificateStatus`?

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs [hunk 6]
```diff
@@ -0,0 +1,611 @@
...
+        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Sphere.DeploymentResource>> GetAsync(string deploymentName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        System.Collections.Generic.IAsyncEnumerator<Azure.ResourceManager.Sphere.DeploymentResource> System.Collections.Generic.IAsyncEnumerable<Azure.ResourceManager.Sphere.DeploymentResource>.GetAsyncEnumerator(System.Threading.CancellationToken cancellationToken) { throw null; }
+        System.Collections.Generic.IEnumerator<Azure.ResourceManager.Sphere.DeploymentResource> System.Collections.Generic.IEnumerable<Azure.ResourceManager.Sphere.DeploymentResource>.GetEnumerator() { throw null; }
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
+    }
+    public partial class DeploymentData : Azure.ResourceManager.Models.ResourceData
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs:115]: Could we rename this type to `SphereDeploymentData`?

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs [hunk 7]
```diff
@@ -0,0 +1,611 @@
...
+        public virtual Azure.AsyncPageable<Azure.ResourceManager.Sphere.DeviceResource> GetAllAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Sphere.DeviceResource>> GetAsync(string deviceName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        System.Collections.Generic.IAsyncEnumerator<Azure.ResourceManager.Sphere.DeviceResource> System.Collections.Generic.IAsyncEnumerable<Azure.ResourceManager.Sphere.DeviceResource>.GetAsyncEnumerator(System.Threading.CancellationToken cancellationToken) { throw null; }
+        System.Collections.Generic.IEnumerator<Azure.ResourceManager.Sphere.DeviceResource> System.Collections.Generic.IEnumerable<Azure.ResourceManager.Sphere.DeviceResource>.GetEnumerator() { throw null; }
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
+    }
+    public partial class DeviceData : Azure.ResourceManager.Models.ResourceData
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs:152]: Could we rename this type to `SphereDeviceData`?

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs [hunk 8]
```diff
@@ -0,0 +1,611 @@
...
+        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Sphere.DeviceGroupResource>> GetAsync(string deviceGroupName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        System.Collections.Generic.IAsyncEnumerator<Azure.ResourceManager.Sphere.DeviceGroupResource> System.Collections.Generic.IAsyncEnumerable<Azure.ResourceManager.Sphere.DeviceGroupResource>.GetAsyncEnumerator(System.Threading.CancellationToken cancellationToken) { throw null; }
+        System.Collections.Generic.IEnumerator<Azure.ResourceManager.Sphere.DeviceGroupResource> System.Collections.Generic.IEnumerable<Azure.ResourceManager.Sphere.DeviceGroupResource>.GetEnumerator() { throw null; }
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
+    }
+    public partial class DeviceGroupData : Azure.ResourceManager.Models.ResourceData
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs:178]: Could we rename this type to `SphereDeviceGroupData`?

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs [hunk 9]
```diff
@@ -0,0 +1,611 @@
...
+        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Sphere.ImageResource>> GetAsync(string imageName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        System.Collections.Generic.IAsyncEnumerator<Azure.ResourceManager.Sphere.ImageResource> System.Collections.Generic.IAsyncEnumerable<Azure.ResourceManager.Sphere.ImageResource>.GetAsyncEnumerator(System.Threading.CancellationToken cancellationToken) { throw null; }
+        System.Collections.Generic.IEnumerator<Azure.ResourceManager.Sphere.ImageResource> System.Collections.Generic.IEnumerable<Azure.ResourceManager.Sphere.ImageResource>.GetEnumerator() { throw null; }
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
+    }
+    public partial class ImageData : Azure.ResourceManager.Models.ResourceData
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs:244]: Could we rename this type to `SphereImageData`?

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs [hunk 10]
```diff
@@ -0,0 +1,611 @@
...
+        public virtual Azure.AsyncPageable<Azure.ResourceManager.Sphere.ProductResource> GetAllAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Sphere.ProductResource>> GetAsync(string productName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        System.Collections.Generic.IAsyncEnumerator<Azure.ResourceManager.Sphere.ProductResource> System.Collections.Generic.IAsyncEnumerable<Azure.ResourceManager.Sphere.ProductResource>.GetAsyncEnumerator(System.Threading.CancellationToken cancellationToken) { throw null; }
+        System.Collections.Generic.IEnumerator<Azure.ResourceManager.Sphere.ProductResource> System.Collections.Generic.IEnumerable<Azure.ResourceManager.Sphere.ProductResource>.GetEnumerator() { throw null; }
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
+    }
+    public partial class ProductData : Azure.ResourceManager.Models.ResourceData
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs:286]: Could we rename this type to `SphereProductData`?

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs [hunk 11]
```diff
@@ -0,0 +1,611 @@
...
+        public static Azure.ResourceManager.Sphere.DeviceGroupResource GetDeviceGroupResource(this Azure.ResourceManager.ArmClient client, Azure.Core.ResourceIdentifier id) { throw null; }
+        public static Azure.ResourceManager.Sphere.DeviceResource GetDeviceResource(this Azure.ResourceManager.ArmClient client, Azure.Core.ResourceIdentifier id) { throw null; }
+        public static Azure.ResourceManager.Sphere.ImageResource GetImageResource(this Azure.ResourceManager.ArmClient client, Azure.Core.ResourceIdentifier id) { throw null; }
+        public static Azure.ResourceManager.Sphere.ProductResource GetProductResource(this Azure.ResourceManager.ArmClient client, Azure.Core.ResourceIdentifier id) { throw null; }
+    }
+}
+namespace Azure.ResourceManager.Sphere.Models
+{
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct AllowCrashDumpCollection : System.IEquatable<Azure.ResourceManager.Sphere.Models.AllowCrashDumpCollection>
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs:332]: It is not a good name for a type not inheriting from ArmCollection to end with the word `Collection`.
Could we rename this to `SphereAllowCrashDumpCollectionStatus` maybe? Or take your own choice.

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs [hunk 12]
```diff
@@ -0,0 +1,611 @@
...
+        public static Azure.ResourceManager.Sphere.Models.ProofOfPossessionNonceResponse ProofOfPossessionNonceResponse(string certificate = null, Azure.ResourceManager.Sphere.Models.CertificateStatus? status = default(Azure.ResourceManager.Sphere.Models.CertificateStatus?), string subject = null, string thumbprint = null, System.DateTimeOffset? expiryUtc = default(System.DateTimeOffset?), System.DateTimeOffset? notBeforeUtc = default(System.DateTimeOffset?), Azure.ResourceManager.Sphere.Models.ProvisioningState? provisioningState = default(Azure.ResourceManager.Sphere.Models.ProvisioningState?)) { throw null; }
+        public static Azure.ResourceManager.Sphere.Models.SignedCapabilityImageResponse SignedCapabilityImageResponse(string image = null) { throw null; }
+    }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct CapabilityType : System.IEquatable<Azure.ResourceManager.Sphere.Models.CapabilityType>
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs:367]: This name is quite generic, could we be more specific on this type name?

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs [hunk 13]
```diff
@@ -0,0 +1,611 @@
...
+        public static implicit operator Azure.ResourceManager.Sphere.Models.CapabilityType (string value) { throw null; }
+        public static bool operator !=(Azure.ResourceManager.Sphere.Models.CapabilityType left, Azure.ResourceManager.Sphere.Models.CapabilityType right) { throw null; }
+        public override string ToString() { throw null; }
+    }
+    public partial class CatalogPatch
+    {
+        public CatalogPatch() { }
+        public System.Collections.Generic.IDictionary<string, string> Tags { get { throw null; } }
+    }
+    public partial class CertificateChainResponse
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs:389]: Could we rename this to `SphereCertificateChainResult`?

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs [hunk 14]
```diff
@@ -0,0 +1,611 @@
...
+    {
+        internal CertificateChainResponse() { }
+        public string CertificateChain { get { throw null; } }
+    }
+    public partial class CertificateProperties
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs:394]: Could we rename this type to `SphereCertificateProperties`?

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs [hunk 15]
```diff
@@ -0,0 +1,611 @@
...
+        public static implicit operator Azure.ResourceManager.Sphere.Models.CertificateStatus (string value) { throw null; }
+        public static bool operator !=(Azure.ResourceManager.Sphere.Models.CertificateStatus left, Azure.ResourceManager.Sphere.Models.CertificateStatus right) { throw null; }
+        public override string ToString() { throw null; }
+    }
+    public partial class ClaimDevicesContent
+    {
+        public ClaimDevicesContent(System.Collections.Generic.IEnumerable<string> deviceIdentifiers) { }
+        public System.Collections.Generic.IList<string> DeviceIdentifiers { get { throw null; } }
+    }
+    public partial class CountDeviceResponse : Azure.ResourceManager.Sphere.Models.CountElementsResponse
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs:430]: Shall we change the suffix `Response` to `Result`?

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs [hunk 16]
```diff
@@ -0,0 +1,611 @@
...
+    {
+        internal CountDeviceResponse() { }
+    }
+    public partial class CountElementsResponse
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs:434]: Shall we change the suffix Response to Result?

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs [hunk 17]
```diff
@@ -0,0 +1,611 @@
...
+    public partial class DeviceGroupPatch
+    {
+        public DeviceGroupPatch() { }
+        public Azure.ResourceManager.Sphere.Models.AllowCrashDumpCollection? AllowCrashDumpsCollection { get { throw null; } set { } }
+        public string Description { get { throw null; } set { } }
+        public Azure.ResourceManager.Sphere.Models.OSFeedType? OSFeedType { get { throw null; } set { } }
+        public Azure.ResourceManager.Sphere.Models.RegionalDataBoundary? RegionalDataBoundary { get { throw null; } set { } }
+        public Azure.ResourceManager.Sphere.Models.UpdatePolicy? UpdatePolicy { get { throw null; } set { } }
+    }
+    public partial class DeviceInsight
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs:448]: Could we rename this type to `SphereDeviceInsight`?

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs [hunk 18]
```diff
@@ -0,0 +1,611 @@
...
+        public DevicePatch() { }
+        public string DeviceGroupId { get { throw null; } set { } }
+    }
+    public partial class GenerateCapabilityImageContent
+    {
+        public GenerateCapabilityImageContent(System.Collections.Generic.IEnumerable<Azure.ResourceManager.Sphere.Models.CapabilityType> capabilities) { }
+        public System.Collections.Generic.IList<Azure.ResourceManager.Sphere.Models.CapabilityType> Capabilities { get { throw null; } }
+    }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct ImageType : System.IEquatable<Azure.ResourceManager.Sphere.Models.ImageType>
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs:471]: This name is quite generic, could we rename it to be more specific?

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs [hunk 19]
```diff
@@ -0,0 +1,611 @@
...
+        public static bool operator !=(Azure.ResourceManager.Sphere.Models.ImageType left, Azure.ResourceManager.Sphere.Models.ImageType right) { throw null; }
+        public override string ToString() { throw null; }
+    }
+    public partial class ListDeviceGroupsContent
+    {
+        public ListDeviceGroupsContent() { }
+        public string DeviceGroupName { get { throw null; } set { } }
+    }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct OSFeedType : System.IEquatable<Azure.ResourceManager.Sphere.Models.OSFeedType>
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs:516]: This name is quite generic, could we rename it to be more specific? For instance, maybe simply `SphereOSFeedType`?

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs [hunk 20]
```diff
@@ -0,0 +1,611 @@
...
+        public static bool operator !=(Azure.ResourceManager.Sphere.Models.RegionalDataBoundary left, Azure.ResourceManager.Sphere.Models.RegionalDataBoundary right) { throw null; }
+        public override string ToString() { throw null; }
+    }
+    public partial class SignedCapabilityImageResponse
+    {
+        internal SignedCapabilityImageResponse() { }
+        public string Image { get { throw null; } }
+    }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct UpdatePolicy : System.IEquatable<Azure.ResourceManager.Sphere.Models.UpdatePolicy>
```
Comment (ArcturusZhang) [inline sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs:594]: This name is quite generic, could we rename it to be more specific?

**Review by ArcturusZhang — DISMISSED** [review]
I have some comments on the namings in the generated SDKs.
Please follow this guide to make changes to them: https://github.com/Azure/autorest.csharp/blob/feature/v3/docs/mgmt/polishing.md#rename-mapping
Please note that we only require you to change the names in this SDK, no changes are required to be done in the corresponding spec unless you find it necessary.
Please be sure to also generate the api file every time you change the generated code and you should be able to see the change of APIs intuitively.

File: sdk/sphere/Azure.ResourceManager.Sphere/CHANGELOG.md [hunk 21]
```diff
@@ -0,0 +1,17 @@
+# Release History
+
+## 1.0.0-beta.1 (Unreleased)
```
Comment (ArthurMa1978) [inline sdk/sphere/Azure.ResourceManager.Sphere/CHANGELOG.md:3]: Please specify the desired release date here, like 2023-06-30

File: sdk/sphere/Azure.ResourceManager.Sphere/src/Properties/AssemblyInfo.cs [hunk 22]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Runtime.CompilerServices;
+
+[assembly: Azure.Core.AzureResourceProviderNamespace("Sphere")]
```
Comment (ArthurMa1978) [inline sdk/sphere/Azure.ResourceManager.Sphere/src/Properties/AssemblyInfo.cs:6]: The value here is `Microsoft.AzureSphere`

File: sdk/sphere/Azure.ResourceManager.Sphere/README.md (hunk 1 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/README.md]: added in latest commit

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs (hunk 2 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs]: fixed in latest commit

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs (hunk 4 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs]: fixed in latest commit

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs (hunk 5 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs]: fixed in latest commit

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs (hunk 3 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs]: fixed in latest commit

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs (hunk 8 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs]: fixed in latest commit

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs (hunk 10 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs]: fixed in latest commit

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs (hunk 11 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs]: fixed in latest commit

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs (hunk 12 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs]: fixed in latest commit

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs (hunk 13 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs]: fixed in latest commit

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs (hunk 14 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs]: fixed in latest commit

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs (hunk 15 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs]: fixed in latest commit

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs (hunk 16 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs]: fixed in latest commit

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs (hunk 17 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs]: fixed in latest commit

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs (hunk 18 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs]: fixed in latest commit

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs (hunk 19 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs]: fixed in latest commit

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs (hunk 9 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs]: fixed in latest commit

File: sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs (hunk 20 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/api/Azure.ResourceManager.Sphere.netstandard2.0.cs]: fixed in latest commit

File: sdk/sphere/Azure.ResourceManager.Sphere/src/Properties/AssemblyInfo.cs (hunk 22 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/src/Properties/AssemblyInfo.cs]: fixed in latest commit

File: sdk/sphere/Azure.ResourceManager.Sphere/CHANGELOG.md (hunk 21 above)
Comment (allegradomel) [reply sdk/sphere/Azure.ResourceManager.Sphere/CHANGELOG.md]: fixed in latest commit

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**